classes/userGroup/UserGroup.php

```diff
@@ -3,263 +3,387 @@
+     */
+    protected array $multilingualProps = [
+        'name',
+        'abbrev',
+    ];
+
+    /**
+     * Get the settings table name
+     */
+    public function getSettingsTable(): string
     {
-        $this->setData('roleId', $roleId);
+        return 'user_group_settings';
     }
 
     /**
-     * Get the role path
-     *
-     * @return string Role path
+     * Get the schema name for the model
      */
-    public function getPath()
+    public static function getSchemaName(): ?string
     {
-        return $this->getData('path');
+        return 'userGroup';
     }
 
     /**
-     * Set the role path
-     * $param $path string
+     * Ensure casts are string values.
+     *
+     * @param array $casts
+     * @return array
      */
-    public function setPath($path)
+    protected function ensureCastsAreStringValues($casts): array
     {
-        $this->setData('path', $path);
+        return array_map(fn($cast) => (string) $cast, $casts);
     }
 
     /**
-     * Get the context ID
+     * setAttribute method for multilingual attributes
+     *
+     * @param string $key
+     * @param mixed $value
+     * @return $this
      */
-    public function getContextId(): ?int
+    public function setAttribute($key, $value)
     {
-        return $this->getData('contextId');
+        if (in_array($key, $this->settings)) {
+            // it's a settings attribute
+            if (in_array($key, $this->multilingualProps)) {
+                // and it's a multilingual setting
+                if (is_string($value)) {
+                    $value = $this->localizeNonLocalizedData($value);
+                }
+            }
+            $this->settings[$key] = $value;
+            return $this;
+        }
+
+        return parent::setAttribute($key, $value);
     }
 
     /**
-     * Set the context ID
+     * Override getAttribute to handle multilingual attributes.
      */
-    public function setContextId(?int $contextId): void
+    public function getAttribute($key)
     {
-        $this->setData('contextId', $contextId);
+        if (in_array($key, $this->settings)) {
+            // it's a settings attribute
+            return $this->settings[$key] ?? null;
+        }
+
+        return parent::getAttribute($key);
     }
 
     /**
-     * Get the default flag
+     * Wraps a non localized value with the default locale
      *
-     * @return bool
+     * @param string $value the non localized value
+     * @return array array with the default locale as the key and the value
      */
-    public function getDefault()
+    protected function localizeNonLocalizedData(string $value): array
     {
-        return $this->getData('isDefault');
+        return [Locale::getLocale() => $value];
     }
 
     /**
-     * Set the default flag
-     *
-     * @param bool $isDefault
+     * Define the relationship to StageAssignments
      */
-    public function setDefault($isDefault)
+    public function stageAssignments(): HasMany
```

Comment:
Can you also add the relationship to the StageAssignment? And the same for the UserUserGroup